### PR TITLE
PSScriptAnalyser Suggestions

### PR DESCRIPTION
**Pull request by vedantmgoyal2009**
- Trim Trailing Whitespaces
- `Write-Colors` to `Write-ColorizedLine`
- `Read-Installer-Values` to `Read-InstallerValue`
- `Enter-PR-Parameters` to `Get-PRParameters`
- `Remove-Manifest-Version` to `Remove-Manifest`
- `Write-Locale-Manifests` to `Write-LocaleManifest`

### CODE REVIEW

**Comment by vedantmgoyal2009** [comment]
@Trenly

**Review by jedieaston — CHANGES_REQUESTED** [review]
Can you do Write-Host -> Write-Output? If not, once this is merged I might have bandwidth to do it tomorrow.

File: Tools/YamlCreate.ps1 [hunk 1]
```diff
@@ -329,7 +329,7 @@ Function Request-Installer-Url {
 # Prompts the user to enter installer values
 # Sets the $script:Installers value as an output
 # Returns void
-Function Read-Installer-Values {
+Function Read-InstallerValues {
```
Comment (Trenly) [inline Tools/YamlCreate.ps1:332]:
```suggestion
Function Read-InstallerEntry {
```

File: Tools/YamlCreate.ps1 [hunk 2]
```diff
@@ -744,14 +744,14 @@ Function Read-Installer-Values {
 
     # If there are additional entries, run this function again to fetch the values and add them to the installers array
     if ($AnotherInstaller -eq '0') {
-        Write-Host; Read-Installer-Values
+        Write-Host; Read-InstallerValues
```
Comment (Trenly) [inline Tools/YamlCreate.ps1:747]:
```suggestion
        Write-Host; Read-InstallerEntry
```

File: Tools/YamlCreate.ps1 [hunk 3]
```diff
@@ -744,14 +744,14 @@ Function Read-Installer-Values {
...
     }
 }
 
 # Prompts user for Installer Values using the `Quick Update` Method
 # Sets the $script:Installers value as an output
 # Returns void
-Function Read-Installer-Values-Minimal {
+Function Read-InstallerValues-Minimal {
```
Comment (Trenly) [inline Tools/YamlCreate.ps1:754]:
```suggestion
Function Read-MinimalInstallerEntry {
```

File: Tools/YamlCreate.ps1 [hunk 4]
```diff
@@ -1297,7 +1297,7 @@ Function Read-WinGet-LocaleManifest {
 
 # Requests the user to answer the prompts found in the winget-pkgs pull request template
 # Uses this template and responses to create a PR
-Function Enter-PR-Parameters {
+Function Get-PRParameters {
```
Comment (Trenly) [inline Tools/YamlCreate.ps1:1300]:
```suggestion
Function Get-PRBody {
```

File: Tools/YamlCreate.ps1 [hunk 5]
```diff
@@ -2066,35 +2066,35 @@ if ($OldManifests -and $Option -ne 'NewLocale') {
 # Run the data entry and creation of manifests appropriate to the option the user selected
 Switch ($script:Option) {
     'QuickUpdateVersion' {
-        Read-Installer-Values-Minimal
-        Write-Locale-Manifests
+        Read-InstallerValues-Minimal
```
Comment (Trenly) [inline Tools/YamlCreate.ps1:2069]:
```suggestion
        Read-MinimalInstallerEntry
```

File: Tools/YamlCreate.ps1 [hunk 6]
```diff
@@ -2066,35 +2066,35 @@ if ($OldManifests -and $Option -ne 'NewLocale') {
...
+        Write-LocaleManifest
         Write-Installer-Manifest
         Write-Version-Manifest
     }
 
     'New' {
-        Read-Installer-Values
+        Read-InstallerValues
```
Comment (Trenly) [inline Tools/YamlCreate.ps1:2076]:
```suggestion
        Read-InstallerEntry
```

File: Tools/YamlCreate.ps1 [hunk 7]
```diff
@@ -2317,14 +2317,14 @@ if ($PromptSubmit -eq '0') {
         if (Get-Command 'gh.exe' -ErrorAction SilentlyContinue) {
             # Request the user to fill out the PR template
             if (Test-Path -Path "$PSScriptRoot\..\.github\PULL_REQUEST_TEMPLATE.md") {
-                Enter-PR-Parameters "$PSScriptRoot\..\.github\PULL_REQUEST_TEMPLATE.md"
+                Get-PRParameters "$PSScriptRoot\..\.github\PULL_REQUEST_TEMPLATE.md"
```
Comment (Trenly) [inline Tools/YamlCreate.ps1:2320]:
```suggestion
                Get-PRBody "$PSScriptRoot\..\.github\PULL_REQUEST_TEMPLATE.md"
```

File: Tools/YamlCreate.ps1 [hunk 8]
```diff
@@ -2317,14 +2317,14 @@ if ($PromptSubmit -eq '0') {
...
             } else {
                 while ([string]::IsNullOrWhiteSpace($SandboxScriptPath)) {
                     Write-Host
                     Write-Host -ForegroundColor 'Green' -Object 'PULL_REQUEST_TEMPLATE.md not found, input path'
                     $PRTemplate = Read-Host -Prompt 'PR Template' | TrimString
                 }
-                Enter-PR-Parameters "$PRTemplate"
+                Get-PRParameters "$PRTemplate"
```
Comment (Trenly) [inline Tools/YamlCreate.ps1:2327]:
```suggestion
                Get-PRBody "$PRTemplate"
```

**Review by Trenly — CHANGES_REQUESTED** [review]
Please also make the following changes:
1. Rename `String.Validate` to `Validate-String` or `Test-String`
3. Rename `Write-Installer-Manifest` to `Write-InstallerManifest`
3. Rename `Write-Version-Manifest` to `Write-VersionManifest`
4. Rename `Read-Winget-InstallerManifest` to `Read-InstallerManifest`
5. Rename `Read-Winget-LocaleManifest` to `Read-LocaleManifest`
4. Fill all the empty catch blocks with `Out-Null`

I would also appreciate if you could split it into multiple PR's for each major change; I know that I can look at each individual commit, but I think it would be easier to ensure nothing breaks if we take it one step at a time. Might I suggest separate pr's for each of these:
- Removing whitespace
- Renaming functions
- Converting write-host to write-output
- Making positional parameters named parameters
- Filling empty catch blocks

**Comment by vedantmgoyal2009** [comment]
@Trenly I've been preparing something new, so it would take me some hours.

**Review by jedieaston — APPROVED** [review]

**Comment by Trenly** [comment]
All PSScriptAnalyzer issues have been resolved

**Pull request closed without merging**